services/management/dbaas/components_service.go

```diff
@@ -392,3 +390,90 @@ func setComponent(kc *models.Component, rc *dbaasv1beta1.ChangeComponent) (*mode
+	if component == nil {
+		return false, nil
+	}
+	supported, err := c.versionServiceClient.IsDatabaseVersionSupportedByOperator(ctx, operatorType, operatorVersion, component.DefaultVersion)
+	if !supported {
+		if err != nil {
+			return false, err
+		}
+		// When default version is empty, the latest recommended version is used.
+		component.DefaultVersion = ""
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c componentsService) InstallOperator(ctx context.Context, req *dbaasv1beta1.InstallOperatorRequest) (*dbaasv1beta1.InstallOperatorResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(c.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	parsedPMMVersion, err := goversion.NewVersion(pmmversion.PMMVersion)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	pmmVersion := parsedPMMVersion.Core().String()
+
+	// Just check the version we are asked to update to is present in version service.
+	resp, err := c.versionServiceClient.Matrix(ctx, componentsParams{product: "pmm-server", productVersion: pmmVersion})
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if len(resp.Versions) == 0 {
+		return nil, errors.Errorf("failed to validate operator version")
+	}
+
+	var component *models.Component
+	switch req.OperatorType {
+	case pxcOperator:
+		if _, ok := resp.Versions[0].Matrix.PXCOperator[req.Version]; !ok {
+			return nil, errors.Errorf("requested update to operator version %s is not supported", req.Version)
+		}
+		_, err = c.dbaasClient.InstallXtraDBOperator(ctx, &controllerv1beta1.InstallXtraDBOperatorRequest{
+			KubeAuth: &controllerv1beta1.KubeAuth{
+				Kubeconfig: kubernetesCluster.KubeConfig,
+			},
+			Version: req.Version,
+		})
+		component = kubernetesCluster.PXC
+	case psmdbOperator:
+		if _, ok := resp.Versions[0].Matrix.PSMDBOperator[req.Version]; !ok {
+			return nil, errors.Errorf("requested update to operator version %s is not supported", req.Version)
+		}
+		_, err = c.dbaasClient.InstallPSMDBOperator(ctx, &controllerv1beta1.InstallPSMDBOperatorRequest{
+			KubeAuth: &controllerv1beta1.KubeAuth{
+				Kubeconfig: kubernetesCluster.KubeConfig,
+			},
+			Version: req.Version,
+		})
+		component = kubernetesCluster.Mongod
+	default:
+		return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_NOT_INSTALLED},
+			errors.Errorf("%q is not supported operator", req.OperatorType)
+	}
+	if err != nil {
+		return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_NOT_INSTALLED},
+			status.Errorf(codes.Internal, "failed to install operator: %v", err)
+	}
+
+	// Default version of database could have become unsupported when update was installed.
+	unset, err := c.unsetDefaultVersionIfUnsupported(ctx, req.OperatorType, req.Version, component)
```

Comment:
Did you discuss it with @rnovikovP ?

Reply:
I did not, but it kind of makes sense, doesn't it? When it certain db version gets unsupported by installed operator, we should not advertise to user it's ok to use it.

Reply:
I asked Roma in slack

Reply:
I now return error when unsupported version is detected.